ts/packages/providers/vercel/src/index.ts

```diff
@@ -166,10 +174,48 @@ export class VercelProvider extends BaseAgenticProvider<VercelToolCollection, Ve
    * }
    * ```
    */
-  wrapTools(tools: ComposioTool[], executeTool: ExecuteToolFn): VercelToolCollection {
+  wrapTools(tools: Tool[], executeTool: ExecuteToolFn): VercelToolCollection {
     return tools.reduce((acc, tool) => {
       acc[tool.slug] = this.wrapTool(tool, executeTool);
       return acc;
     }, {} as VercelToolCollection);
   }
+
+  /**
+   * Transform MCP URL response into Vercel-specific format.
+   * Vercel uses the standard format by default.
+   *
+   * @param data - The MCP URL response data
+   * @param serverName - Name of the MCP server
+   * @param connectedAccountIds - Optional array of connected account IDs
+   * @param userIds - Optional array of user IDs
+   * @param toolkits - Optional array of toolkit names
+   * @returns Standard MCP server response format
+   */
+  transformMcpResponse(
+    data: McpUrlResponse,
+    serverName: string,
+    connectedAccountIds?: string[],
+    userIds?: string[],
+    toolkits?: string[]
+  ): McpServerGetResponse {
+    // Vercel uses the standard format
+    if (connectedAccountIds?.length && data.connected_account_urls) {
+      return data.connected_account_urls.map((url: string, index: number) => ({
+        url: new URL(url),
+        name: `${serverName}-${connectedAccountIds[index]}`,
```

Comment:
I can see the id/name is changed everywhere, ideally move this logic to the core/mcp so that peopel defining the provider doesn't have to write this. This is an internal logic leaking here. If you can do this within the mcp, you can avoid repeated code everywhere in every provider.